JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/derivation/DerivationClient.java

```diff
@@ -123,6 +123,12 @@ public List<String> bulkCreateDerivations(List<List<String>> entitiesList, List<
 		// add timestamp to each derivation
 		this.sparqlClient.addTimeInstance(derivations);
 
+		// validate derivations if more than one derivations are marked up
+		// this is to prevent the potential circular dependencies in the markup added but not detected at creation
+		if (derivations.size() > 1) {
```

Comment:
(!derivations.size().isEmpty()) is better

Reply:
(!derivations.isEmpty()) even better ;-)

Reply:
This is changed as suggested in the latest commits.